### DISCUSSION TITLE
Failure to install ganche from vsCode extension

### DISCUSSION THREAD

**Discussion by axelrg** [ConsenSys-archive/truffle#5719] [Installation]
![image](https://user-images.githubusercontent.com/63260510/202904572-970bac69-9cfc-40a0-8168-a01891708587.png)
Apparently only ganache is missing but when i try to install it from the requirements page it gives me this error
![image](https://user-images.githubusercontent.com/63260510/202905004-772ef3c3-1925-4a49-835d-47ee27f1053b.png)
In the line where it mentions "node-gyp-build" it basically means that that line is not recognized as any internal or external command
Thanks for your help in advanced.

**Comment by apdan510** [comment]
@axelrg could you please open a fresh terminal and try a plain npm ganache install with `npm install -g ganache` and share the return output for us? 

Thank you,
Alex | Truffle Support | Consensys

**Reply by axelrg** [reply]
![image](https://user-images.githubusercontent.com/63260510/204089120-71970588-24d0-45ff-b7f6-b403b26275ab.png)
Basically gives the same error.

This is the returned log:
0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install',
0 verbose cli   '-g',
0 verbose cli   'ganache'
0 verbose cli ]
1 info using npm@8.3.1
2 info using node@v16.14.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:C:\Program Files\nodejs\node_modules\npm\npmrc Completed in 2ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:C:\Users\Axel\.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:C:\Users\Axel\AppData\Roaming\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:validate Completed in 0ms
15 timing config:load:credentials Completed in 1ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 9ms
18 timing npm:load:configload Completed in 9ms
19 timing npm:load:setTitle Completed in 0ms
20 timing config:load:flatten Completed in 2ms
21 timing npm:load:display Completed in 3ms
22 verbose logfile C:\Users\Axel\AppData\Local\npm-cache\_logs\2022-11-26T12_31_19_776Z-debug-0.log
23 timing npm:load:logFile Completed in 4ms
24 timing npm:load:timers Completed in 0ms
25 timing npm:load:configScope Completed in 0ms
26 timing npm:load Completed in 16ms
27 timing arborist:ctor Completed in 0ms
28 silly logfile start cleaning logs, removing 2 files
29 timing idealTree:init Completed in 3ms
30 timing idealTree:userRequests Completed in 1ms
31 silly idealTree buildDeps
32 silly fetch manifest ganache@*
33 http fetch GET 200 https://registry.npmjs.org/ganache 917ms (cache revalidated)
34 silly placeDep ROOT ganache@7.5.0 OK for:  want: *
35 silly fetch manifest bufferutil@4.0.5
36 silly fetch manifest utf-8-validate@5.0.7
37 http fetch GET 200 https://registry.npmjs.org/utf-8-validate 592ms (cache revalidated)
38 http fetch GET 200 https://registry.npmjs.org/bufferutil 782ms (cache revalidated)
39 timing idealTree:#root Completed in 1709ms
40 timing idealTree:buildDeps Completed in 1709ms
41 timing idealTree:fixDepFlags Completed in 0ms
42 timing idealTree Completed in 1715ms
43 timing reify:loadTrees Completed in 1717ms
44 timing reify:diffTrees Completed in 1ms
45 silly reify moves {}
46 timing reify:retireShallow Completed in 0ms
47 timing reify:createSparse Completed in 1ms
48 timing reifyNode:node_modules/ganache Completed in 1258ms
49 timing arborist:ctor Completed in 0ms
50 timing reify:diffTrees Completed in 1ms
51 timing reify:createSparse Completed in 2ms
52 timing reify:loadShrinkwraps Completed in 1286ms
53 timing reifyNode:node_modules/ganache Completed in 1081ms
54 timing arborist:ctor Completed in 0ms
55 timing reify:diffTrees Completed in 1ms
56 timing reify:loadBundles Completed in 1110ms
57 silly audit bulk request {
57 silly audit   'node-gyp-build': [ '4.3.0', '4.4.0' ],
57 silly audit   '@types/node': [ '17.0.0' ],
57 silly audit   'base64-js': [ '1.5.1' ],
57 silly audit   brorand: [ '1.1.0' ],
57 silly audit   buffer: [ '6.0.3' ],
57 silly audit   bufferutil: [ '4.0.5' ],
57 silly audit   catering: [ '2.1.0' ],
57 silly audit   elliptic: [ '6.5.4' ],
57 silly audit   'bn.js': [ '4.12.0' ],
57 silly audit   'hash.js': [ '1.1.7' ],
57 silly audit   'hmac-drbg': [ '1.0.1' ],
57 silly audit   ieee754: [ '1.2.1' ],
57 silly audit   inherits: [ '2.0.4' ],
57 silly audit   'is-buffer': [ '2.0.5' ],
57 silly audit   'minimalistic-assert': [ '1.0.1' ],
57 silly audit   'minimalistic-crypto-utils': [ '1.0.1' ],
57 silly audit   'napi-macros': [ '2.0.0' ],
57 silly audit   'node-addon-api': [ '2.0.2' ],
57 silly audit   'queue-microtask': [ '1.2.3' ],
57 silly audit   'queue-tick': [ '1.0.0' ],
57 silly audit   'readable-stream': [ '3.6.0' ],
57 silly audit   'safe-buffer': [ '5.2.1' ],
57 silly audit   string_decoder: [ '1.3.0' ],
57 silly audit   'utf-8-validate': [ '5.0.7' ],
57 silly audit   'util-deprecate': [ '1.0.2' ],
57 silly audit   ganache: [ '7.5.0' ],
57 silly audit   emittery: [ '0.10.0' ],
57 silly audit   keccak: [ '3.0.2' ],
57 silly audit   leveldown: [ '6.1.0' ],
57 silly audit   'abstract-leveldown': [ '7.2.0' ],
57 silly audit   'level-concat-iterator': [ '3.1.0' ],
57 silly audit   'level-supports': [ '2.1.0' ],
57 silly audit   secp256k1: [ '4.0.3' ],
57 silly audit   '@trufflesuite/bigint-buffer': [ '1.1.10' ],
57 silly audit   '@types/bn.js': [ '5.1.0' ],
57 silly audit   '@types/lru-cache': [ '5.1.1' ],
57 silly audit   '@types/seedrandom': [ '3.0.1' ]
57 silly audit }
58 timing reifyNode:node_modules/ganache/node_modules/utf-8-validate Completed in 33ms
59 timing reifyNode:node_modules/ganache/node_modules/bufferutil Completed in 32ms
60 timing reify:unpack Completed in 33ms
61 timing reify:unretire Completed in 0ms
62 timing build:queue Completed in 2ms
63 timing build:link:node_modules/ganache/node_modules/@trufflesuite/bigint-buffer/node_modules/node-gyp-build Completed in 6ms
64 timing build:link:node_modules/ganache Completed in 6ms
65 timing build:link Completed in 6ms
66 info run @trufflesuite/bigint-buffer@1.1.10 install node_modules/ganache/node_modules/@trufflesuite/bigint-buffer node-gyp-build || echo "Couldn't build bindings. Non-native version used."
67 info run bufferutil@4.0.5 install node_modules/ganache/node_modules/bufferutil node-gyp rebuild
68 info run keccak@3.0.2 install node_modules/ganache/node_modules/keccak node-gyp-build || exit 0
69 info run leveldown@6.1.0 install node_modules/ganache/node_modules/leveldown node-gyp-build
70 info run secp256k1@4.0.3 install node_modules/ganache/node_modules/secp256k1 node-gyp-build || exit 0
71 info run utf-8-validate@5.0.7 install node_modules/ganache/node_modules/utf-8-validate node-gyp rebuild
72 info run leveldown@6.1.0 install { code: 1, signal: null }
73 info run keccak@3.0.2 install { code: 0, signal: null }
74 timing build:run:install:node_modules/ganache/node_modules/keccak Completed in 13ms
75 info run secp256k1@4.0.3 install { code: 0, signal: null }
76 timing build:run:install:node_modules/ganache/node_modules/secp256k1 Completed in 9ms
77 info run @trufflesuite/bigint-buffer@1.1.10 install { code: 0, signal: null }
78 timing build:run:install:node_modules/ganache/node_modules/@trufflesuite/bigint-buffer Completed in 142ms
79 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 265ms
80 timing auditReport:getReport Completed in 267ms
81 silly audit report {}
82 timing auditReport:init Completed in 0ms
83 timing reify:audit Completed in 268ms
84 info run utf-8-validate@5.0.7 install { code: 1, signal: null }
85 verbose reify failed optional dependency C:\Users\Axel\AppData\Roaming\npm\node_modules\ganache\node_modules\utf-8-validate
86 silly reify mark deleted [
86 silly reify   'C:\\Users\\Axel\\AppData\\Roaming\\npm\\node_modules\\ganache\\node_modules\\utf-8-validate'
86 silly reify ]
87 timing build:run:install:node_modules/ganache/node_modules/utf-8-validate Completed in 271ms
88 info run bufferutil@4.0.5 install { code: 1, signal: null }
89 verbose reify failed optional dependency C:\Users\Axel\AppData\Roaming\npm\node_modules\ganache\node_modules\bufferutil
90 silly reify mark deleted [
90 silly reify   'C:\\Users\\Axel\\AppData\\Roaming\\npm\\node_modules\\ganache\\node_modules\\bufferutil'
90 silly reify ]
91 timing build:run:install:node_modules/ganache/node_modules/bufferutil Completed in 283ms
92 timing reify:rollback:createSparse Completed in 322ms
93 timing reify:rollback:retireShallow Completed in 0ms
94 timing command:install Completed in 4511ms
95 verbose stack Error: command failed
95 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:64:27)
95 verbose stack     at ChildProcess.emit (node:events:520:28)
95 verbose stack     at maybeClose (node:internal/child_process:1092:16)
95 verbose stack     at Socket.<anonymous> (node:internal/child_process:451:11)
95 verbose stack     at Socket.emit (node:events:520:28)
95 verbose stack     at Pipe.<anonymous> (node:net:687:12)
96 verbose pkgid leveldown@6.1.0
97 verbose cwd C:\Windows\system32
98 verbose Windows_NT 10.0.19044
99 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "ganache"
100 verbose node v16.14.0
101 verbose npm  v8.3.1
102 error code 1
103 error path C:\Users\Axel\AppData\Roaming\npm\node_modules\ganache\node_modules\leveldown
104 error command failed
105 error command C:\Windows\system32\cmd.exe /d /s /c node-gyp-build
106 error "node-gyp-build" no se reconoce como un comando interno o externo,
106 error programa o archivo por lotes ejecutable.
107 verbose exit 1
108 timing npm Completed in 4757ms
109 verbose unfinished npm timer reify 1669465880019
110 verbose unfinished npm timer reify:build 1669465884176
111 verbose unfinished npm timer build 1669465884177
112 verbose unfinished npm timer build:deps 1669465884177
113 verbose unfinished npm timer build:run:install 1669465884185
114 verbose unfinished npm timer build:run:install:node_modules/ganache/node_modules/leveldown 1669465884196
115 verbose code 1
116 error A complete log of this run can be found in:
116 error     C:\Users\Axel\AppData\Local\npm-cache\_logs\2022-11-26T12_31_19_776Z-debug-0.log

Thanks for your help

**Comment by apdan510** [comment]
Could you try to reinstall node with this installer? https://nodejs.org/dist/v18.12.1/node-v18.12.1-x86.msi
please accept and install the native build tools option at the end. 

<img width="491" alt="image" src="https://user-images.githubusercontent.com/40121410/205126133-c080b6d1-55c3-454e-94ed-8cc6c58be021.png">




Could you try to open an administrative cmdline or powershell and installing ganache globally and let us know what happens?

`npm install -g ganache`
